### PR TITLE
Fix type violations due to missing glint configuration

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Many of the type errors that I could see were due to missing the template-imports glint environment.
This has been added.

### CODE REVIEW

File: packages/ember-headless-form/src/components/headless-form.gts [hunk 1]
```diff
@@ -187,9 +188,11 @@ export default class HeadlessFormComponent<
   DATA extends UserData,
   SUBMISSION_VALUE
 > extends Component<HeadlessFormComponentSignature<DATA, SUBMISSION_VALUE>> {
+  // we cannot use (modifier "on") directly in the template due to https://github.com/emberjs/ember.js/issues/19869
+  on = on;
```
Comment (simonihmig) [inline packages/ember-headless-form/src/components/headless-form.gts:192]: Oh, right. I was too eager to remove this :+1:

File: packages/ember-headless-form/src/components/headless-form.gts [hunk 2]
```diff
@@ -1,12 +1,13 @@
 import Component from '@glimmer/component';
 import { tracked } from '@glimmer/tracking';
 import { assert, warn } from '@ember/debug';
-import { hash } from '@ember/helper';
+// @ts-ignore hash is missing from ember types
```
Comment (simonihmig) [inline packages/ember-headless-form/src/components/headless-form.gts:4]: But linting was happy with this file. IIRC `modifier` was actually missing, not? But somehow it was not needed, the template was still working for me locally without the import 🤔

**Review by simonihmig — COMMENTED** [review]
Thanks so much for looking into this!

CI shows some new issues though, any idea? https://github.com/CrowdStrike/ember-headless-form/actions/runs/4439238826/jobs/7791450494?pr=90#step:5:27

File: packages/ember-headless-form/src/components/headless-form.gts (hunk 2 above)
Comment (NullVoxPopuli) [reply packages/ember-headless-form/src/components/headless-form.gts:4]: yeah, it's a built-in, but not yet for types -- I think there is an open issue about it and it came up in discord a couple days ago

**Comment by NullVoxPopuli** [comment]
>  some new issues though, any idea?

potentially! 

it's possible something got messed up with this code: https://github.com/NullVoxPopuli/rollup-plugin-glimmer-template-tag/blob/main/packages/rollup-plugin-glimmer-template-tag/src/rollup-plugin.js#L90 -- it relies on the consumer's babel config to be able to parse correctly

**Comment by NullVoxPopuli** [comment]
Looks like:
```
Unsafe dynamic component: HeadlessFormFieldComponent in $TMPDIR/embroider/fd7bfa/node_modules/.pnpm/file+packages+ember-headless-form_yj3yrsfg6wbq6iy2hki53xivum/node_modules/ember-headless-form/dist/components/headless-form.js
```

**Comment by simonihmig** [comment]
I'll merge this as-is, and try to fix the remaining issues in the main branch!
Thanks!

**Pull request merged**